### ISSUE TITLE
[ChipGroup] Chip overflow indicator

### ISSUE THREAD

**Issue by CodeRedDev** [material-components/material-components-android#696]
Currently the `ChipGroup` supports the `singleLine` attribute with a `HorizontalScrollView` so that the chips will always fit in a scrollable list. Unfortunately for some use cases this behavior is not really nice.

For example if you have a list of filters that can be selected on an extra screen and will be displayed in a single line `ChipGroup` somewhere on the main screen. The user wouldn't really profit from a scrollable single line that will only support viewing the filters. Instead the ChipGroup should display all items till it reaches its space limit and display an indicator (like "...") that lets the user know that there are more filters set.

I'm trying to get this by calculating the container width and the childrens width to decide which chips to remove but it would be nice to have an overflow behavior attribute as a feature that will automatically do this by itself.

It could look like:
![](https://i.ibb.co/CMDWsDJ/Screenshot-2019-10-22-at-11.jpg)

**Comment by mwajeeh** [comment]
Would love this. I need to show +X on the last chip so that user know there are more x chips.

**Comment by stefan-niedermann** [comment]
> I'm trying to get this by calculating the container width and the childrens width to decide which chips to remove

@CodeRedDev did you achieve this? Can you share the code for your implementation? I would like to adapt it :slightly_smiling_face:

**Comment by CodeRedDev** [comment]
@stefan-niedermann Yes I achieved a solution but unfortunately I do not have access to it anymore cause it's in my former employer's code base. But yes it works pretty good when you get the containers dimensions and the childrens to decide if you have to add this indicator chip. Be aware though that it could sometimes look a little "jumpy" as the calculations are done after the view tree already was layout or drawn.

**Comment by utamilah** [comment]
@CodeRedDev (or anyone) do you know if there's a JavaScript/TypeScript version of this solution? I'm trying to achieve this exact same effect on an Angular project with Angular material. Thanks!
